src/main/java/com/impossibl/postgres/jdbc/PGResultSet.java

```diff
@@ -874,6 +874,9 @@ public Object getObject(int columnIndex, Map<String, Class<?>> map) throws SQLEx
 
     Class<?> targetType = mapGetType(type, map, statement.connection);
 
+    if (statement.connection.isStrictMode() && InputStream.class.equals(targetType))
```

Comment:
It assumes that BINARY and VARBINARY maps to two different database types -- we use "bytea" for both of course.

Ideally getObject(int) shouldn't be used for byte/input streams, but is that really tested ;) So, it is really database + JDBC driver specific what is done.

The reason I did it this way is that assuming InputStream would hit more use-cases for getObject(int) than byte[]. But we need the latter in some way.

Of course this needs documentation once we decide how to get those 4 test cases passing